app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,23 @@
+class SessionsController < ApplicationController
+  def new
+  end
+
+  # POST /login
+  def create
+    user = User.find_by_username(params[:username]).first
+    # If the user exists AND the password entered is correct.
+    if user && user.authenticate(params[:password])
+      # Save the user id
+      session[:user_id] = user.id
+      redirect_to '/'
+    else
```

Comment:
should have flash[:notice] and flash[:error] messages for successful login and unsuccessful login